src/main/haskell/language-kore/src/Data/Kore/Proof/Proof.hs

```diff
@@ -0,0 +1,421 @@
+{-|
+Module      : Data.Kore.Proof.Proof
+Description : Tree-based proof system, which can be
+              hash-consed into a list-based one.
+Copyright   : (c) Runtime Verification, 2018
+License     : UIUC/NCSA
+Maintainer  : phillip.harris@runtimeverification.com
+Stability   : experimental
+Portability : portable
+-}
+
+{-# LANGUAGE AllowAmbiguousTypes       #-}
+{-# LANGUAGE ConstraintKinds           #-}
+{-# LANGUAGE DeriveFoldable            #-}
+{-# LANGUAGE DeriveFunctor             #-}
+{-# LANGUAGE DeriveGeneric             #-}
+{-# LANGUAGE DeriveTraversable         #-}
+{-# LANGUAGE FlexibleContexts          #-}
+{-# LANGUAGE FlexibleInstances         #-}
+{-# LANGUAGE FunctionalDependencies    #-}
+{-# LANGUAGE LambdaCase                #-}
+{-# LANGUAGE MultiParamTypeClasses     #-}
+{-# LANGUAGE NoMonomorphismRestriction #-}
+{-# LANGUAGE PatternSynonyms           #-}
+{-# LANGUAGE Rank2Types                #-}
+{-# LANGUAGE TypeApplications          #-}
+{-# LANGUAGE TypeFamilies              #-}
+{-# LANGUAGE TypeSynonymInstances      #-}
+{-# OPTIONS_GHC -Wno-unused-matches    #-}
+{-# OPTIONS_GHC -Wno-name-shadowing    #-}
+{-# OPTIONS_GHC -Wno-incomplete-patterns  #-}
+
+
+module Data.Kore.Proof.Proof where
```

Comment:
Again, do you want to export everything from this module? If so, you need to document everything :-D

Reply:
Literally everything is useful there, yes.